pkg/controller/master/upgrade/job_controller.go

```diff
@@ -159,23 +160,30 @@ func (h *jobHandler) syncNodeJob(job *batchv1.Job) (*batchv1.Job, error) {
 		}
 	}
 
-	machine, err := h.machineClient.Get(rancherMachineNamespace, machineName, metav1.GetOptions{})
-	if err != nil {
-		return job, err
+	// find machine plan secret
+	secrets, err := h.secretClient.List(rancherPlanSecretNamespace, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", rancherPlanSecretMachineLabel, machineName),
+		FieldSelector: fmt.Sprintf("type=%s", rancherPlanSecretType),
+	})
+
+	if err != nil || len(secrets.Items) != 1 {
+		return job, nil
```

Comment:
Is this intentional not to return `err`?
```suggestion
	if err != nil || len(secrets.Items) != 1 {
		return job, err
```

Reply:
If the length of `secrets.Items` is more than 1, do we need to return err? If yes, we may need to separate this block.

Reply:
No, it's a programming error.
An error or len(secrets) != 1 should both return errrors. Will fix it.

Reply:
@FrankYang0529 yes, need to create an error when len(items) != 1.

Reply:
Fixed, please see https://github.com/harvester/harvester/compare/577b8d4214bf3904cabec39472a077bdeedfc97d..324d08231c8fa1e8061ca9e286899147deb9c5c5